### PR TITLE
Port textanswer review quick view to TypeScript

### PR DESCRIPTION
**Pull request by niklasmohrin**
Before working on the textanswer flagging, I wanted to get this done first.
This ports `quick-review.js` to TypeScript and replaces all uses of jQuery with standard DOM APIs.
I will also apply some refactorings and renames where I see fit.

### CODE REVIEW

File: evap/static/ts/src/quick-review-slider.ts [hunk 1]
```diff
@@ -0,0 +1,410 @@
+// This file accompanies the "quick" view in the textanswer review site. The quick view presents textanswers in a
+// _slider_ that is made up of three layers - answer, question and questionnaire. One answer is shown at a time, it is
+// _active_. The slider contains a form that has multiple submit buttons, one for each review decision, one for revoking
+// a previous decision and one for editing the answer. If a review decision is made, the next answer slides in from the
+// right. At the end, there is one _alert slide_ that summarises the current state of affairs regarding textanswer
```
Comment (janno42) [inline evap/static/ts/src/quick-review-slider.ts:5]:
```suggestion
// right. At the end, there is one _alert slide_ that summarizes the current state of affairs regarding textanswer
```

**Review by janno42 — APPROVED** [review]
✔️ Meets requirements
✔️ UI functionality checked

File: evap/static/ts/src/quick-review-slider.ts [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+    // Sliding
+    //
+    public startOver = (where: StartOverWhere) => {
+        const decided = this.slider.querySelectorAll<HTMLElement>(`[data-layer="${Layer.Answer}"][data-review]`);
+        const undecided = this.slider.querySelectorAll<HTMLElement>(
+            `[data-layer="${Layer.Answer}"]:not([data-review])`,
+        );
+        this.answerSlides = Array.from(decided).concat(Array.from(undecided));
+
+        const startIndex = where === StartOverWhere.Undecided && undecided.length > 0 ? decided.length : 0;
```
Comment (He3lixxx) [inline evap/static/ts/src/quick-review-slider.ts:355]: I'd prefer some parentheses here (I don't have an operator precedence table in my head).

File: evap/static/ts/src/quick-review-slider.ts [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+                activeInParentLayer =
+                    findPreviousElementSibling(nextActiveElement, `[data-layer="${layer - 1}"]`) ?? undefined;
+                activeInParentLayer = activeInParentLayer as HTMLElement | undefined;
+            }
+            this.slideLayer(layer - 1, direction, activeInParentLayer);
+        }
+
+        if (nextActiveElement) {
+            // First, translate nextActiveElement to the left or right and
+            // then, in the next frame, move it back to the middle
```
Comment (He3lixxx) [inline evap/static/ts/src/quick-review-slider.ts:404]: I think we want to have a bit more explanation here. Especially: The animation takes more than one browser frame. Why is it okay to change the class in the next frame?

**Review by He3lixxx — COMMENTED** [review]
Code looks mostly good (but I was very distracted during review, so I might have missed a lot)

File: evap/staff/templates/quick-review.js [hunk 4]
```diff
@@ -1,3 +1,5 @@
+// TODO: DELETE ME
```
Comment (He3lixxx) [inline evap/staff/templates/quick-review.js:1]: keep in mind

**Review by He3lixxx — APPROVED** [review]
UI seems to work reliably

File: evap/static/ts/src/quick-review-slider.ts (hunk 2 above)
Comment (niklasmohrin) [reply evap/static/ts/src/quick-review-slider.ts:355]: prettier says "but you should" :/ I moved the condition into another variable

**Pull request merged**